File: templates/tests-java/BREACH-004.java
package compliance;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class DataResidencyTest {
    @Test
    void testRejectForeignDataResidency() {
        Config config = Config.load();
        String residency = config.getInfrastructure().getDataResidency();
        assertTrue(residency.equals("Local") || residency.equals("Hybrid"),
            "Data residency must be Local or Hybrid, got " + residency);
    }
}

File: templates/tests-go/BREACH-004.go
package compliance

import (
	"testing"
)

func TestRejectForeignDataResidency(t *testing.T) {
	config := GetConfig()
	residency := config.Infrastructure.DataResidency
	if residency != "Local" && residency != "Hybrid" {
		t.Errorf("Data residency must be Local or Hybrid, got %s", residency)
	}
}